include/ebpf_api.h

```diff
@@ -131,60 +140,124 @@ extern "C"
         _Out_ ebpf_handle_t* handle);
 
     /**
-     * @brief Find an element in an eBPF map.
-     * @param[in] handle Handle to eBPF map.
-     * @param[in] key_size Size of the key buffer.
+     * @brief Create an eBPF map with input parameters.
+     *
+     * @param[in] type Map type.
+     * @param[in] key_size Key size.
+     * @param[in] value_size Value size.
+     * @param[in] max_entries Maximum number of entries in the map.
+     * @param[in] map_flags This is reserved and should be 0.
+     * @param[out] map_fd fd for the created map. Caller needs to
+     *  call _close() on the returned fd when done.
+     *
+     * @retval EBPF_SUCCESS Map created successfully.
+     * @retval EBPF_ERROR_NOT_SUPPORTED Unsupported map type.
+     * @retval EBPF_INVALID_ARGUMENT One or more parameters are incorrect.
+     */
+    _Success_(return == EBPF_SUCCESS) ebpf_result_t ebpf_create_map(
+        ebpf_map_type_t map_type,
+        uint32_t key_size,
+        uint32_t value_size,
+        uint32_t max_entries,
+        uint32_t map_flags,
+        _Out_ fd_t* map_fd);
+
+    /**
+     * @brief Create an eBPF map with input parameters.
+     *
+     * @param[in] type Map type.
+     * @param[in] name Optionally, the map name.
+     * @param[in] key_size Key size.
+     * @param[in] value_size Value size.
+     * @param[in] max_entries Maximum number of entries in the map.
+     * @param[in] map_flags This is reserved and should be 0.
+     * @param[out] map_fd fd for the created map. Caller needs to
+     *  call _close() on the returned fd when done.
+     *
+     * @retval EBPF_SUCCESS Map created successfully.
+     * @retval EBPF_ERROR_NOT_SUPPORTED Unsupported map type.
+     * @retval EBPF_INVALID_ARGUMENT One or more parameters are incorrect.
+     */
+    _Success_(return == EBPF_SUCCESS) ebpf_result_t ebpf_create_map_name(
+        ebpf_map_type_t type,
+        _In_opt_z_ const char* name,
+        uint32_t key_size,
+        uint32_t value_size,
+        uint32_t max_entries,
+        uint32_t map_flags,
+        _Out_ fd_t* map_fd);
+
+    /**
+     * @brief Lookup an element in an eBPF map.
+     *  For singleton map, return the value for the given key.
+     *  For per-cpu map, return aggregate value across all CPU.
+     *
+     * @param[in] map_fd File descriptor for the eBPF map.
      * @param[in] key Pointer to buffer containing key.
-     * @param[in] value_size Size of the value buffer.
      * @param[out] value Pointer to buffer that contains value on success.
+     *
+     * @retval Status of the operation.
      */
-    uint32_t
-    ebpf_api_map_find_element(
-        ebpf_handle_t handle, uint32_t key_size, const uint8_t* key, uint32_t value_size, uint8_t* value);
+    _Success_(return == EBPF_SUCCESS) ebpf_result_t
```

Comment:
```suggestion
    ebpf_result_t
```
ebpf_result_t is defined with `_Return_type_success_(return == EBPF_SUCCESS)` as part of the typedef so this is not needed